### ISSUE TITLE
Cannot jump to the desired position if multiple windows contain the same content

### ISSUE THREAD

**Issue by rockyzhang24** [ggandor/lightspeed.nvim#142]
For example, I totally opened three windows and the contents of each one are the same. 

<img width="1715" alt="image" src="https://user-images.githubusercontent.com/11582667/162327302-18a4dd52-a710-4670-b1b7-1a5e06d1034f.png">

My cursor is located at line 126 in the leftmost window. I want to jump to `itemIndex++`, line 132, in the middle window. So I press `gs` followed by `it` 

<img width="1715" alt="image" src="https://user-images.githubusercontent.com/11582667/162327526-2e85f8dc-b213-4e04-8f26-cf6d0a524996.png">

You can see that all these three windows have the same beacons. Then I press `H`. My cursor jumps to line 132 in the rightmost window, instead of the middle one. 

<img width="1715" alt="image" src="https://user-images.githubusercontent.com/11582667/162327675-0395b091-a38a-4823-8ff6-0c4ac49b692f.png">

Next for the further test, I move the contents in the rightmost window upward a little bit. Now I press `gs` followed by `it, and they still have the same beacons and I still cannot jump to the middle window. 

<img width="1715" alt="image" src="https://user-images.githubusercontent.com/11582667/162328078-f608baa3-f889-4a61-91c7-af0ea7a0589e.png">

Thank you so much.

**Comment by rockyzhang24** [comment]
Hello @ggandor Any workaround for this? Thanks a lot.

**Comment by ggandor** [comment]
This is very tricky, because extmarks can only be set for a buffer, and not for a window. So if you open multiple windows with the same content, then you will jump kinda "randomly" to one of them (depends on the order in `winlayout`). That said, I don't see why this is such a huge problem - why would you want to jump to the same buffer position in another window?

**Comment by rockyzhang24** [comment]
I see. Thanks for the explanation. 

> That said, I don't see why this is such a huge problem - why would you want to open more than two windows with the exact same content in the first place, and jump back and forth between them?

For example, sometimes we may open three windows with the same file, one window for editing, and the other two for reference. Look at my last screenshot above please. I use the left window for editing, the middle window and the right window to refer different sections. However, as long as the middle window and the right window have same lines even if they are in a different position, we cannot jump to the window we want. 

Thank you.

**Comment by ggandor** [comment]
Yeah, I see. If there's even some overlap between two or more windows, it can prevent deterministic cross-window jumping. I don't see how this could be solved... I'll think about it though.
